### PR TITLE
CLOUD-651 fix: smartupdate starts after pod deletion

### PR DESCRIPTION
**Pull request by pooknull**
[CLOUD-651](https://jira.percona.com/browse/CLOUD-651) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/CLOUD-651

Fix statefulSet change detection logic

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| version-service | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/808/commits/8da0604bc1b4ebc88a4b8973795d17dc6d395dfd
image: `perconalab/percona-server-mongodb-operator:PR-808-8da0604b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | failed |
| version-service | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/808/commits/5a40045f878670f46d00074d5ac8279bd88d1efa
image: `perconalab/percona-server-mongodb-operator:PR-808-5a40045f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/808/commits/5a40045f878670f46d00074d5ac8279bd88d1efa
image: `perconalab/percona-server-mongodb-operator:PR-808-5a40045f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | failed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/808/commits/4dabc37ae657207b576a6dba4564ff815447b893
image: `perconalab/percona-server-mongodb-operator:PR-808-4dabc37a`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| storage | passed |
| data-sharded | passed |
| self-healing | passed |
| non-voting | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/808/commits/4dabc37ae657207b576a6dba4564ff815447b893
image: `perconalab/percona-server-mongodb-operator:PR-808-4dabc37a`

File: pkg/controller/perconaservermongodb/smart.go [hunk 1]
```diff
@@ -258,3 +269,15 @@ func (r *ReconcilePerconaServerMongoDB) waitPodRestart(cr *api.PerconaServerMong
 
 	return errors.New("reach pod wait limit")
 }
+
+func isSfsChanged(sfs *appsv1.StatefulSet, podList *corev1.PodList) (bool, error) {
```
Comment (egegunes) [inline pkg/controller/perconaservermongodb/smart.go:273]: we don't return error anywhere in the function

**Review by egegunes — CHANGES_REQUESTED** [review]

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/808/commits/1ec87008d2d1f21a85c31cc409d3f9f081e010db
image: `perconalab/percona-server-mongodb-operator:PR-808-1ec87008`

**Review by delgod — APPROVED** [review]

**Pull request merged**